File: src/main/java/com/beat/domain/performance/application/dto/modify/cast/CastModifyResponse.java
package com.beat.domain.performance.application.dto.modify.cast;

public record CastModifyResponse(Long castId,
								 String castName,
								 String castRole,
								 String castPhoto) {

	public static CastModifyResponse of(Long castId, String castName, String castRole, String castPhoto) {
		return new CastModifyResponse(castId, castName, castRole, castPhoto);
	}
}

File: apps/api/src/main/kotlin/com/beat/apps/api/performance/api/response/CastModifyResponse.kt
package com.beat.apps.api.performance.api.response

import com.beat.application.frontoffice.performance.CastResult
import com.beat.apps.api.web.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class CastModifyResponse private constructor(
    val castId: Long?,
    val castName: String?,
    val castRole: String?,
    @field:CdnImageUrl val castPhoto: String?,
) {
    companion object {
        fun from(result: CastResult): CastModifyResponse = CastModifyResponse(result.id, result.name, result.role, result.photo)
    }
}
